community/kernel/src/main/java/org/neo4j/graphdb/ResourceIterator.java

```diff
@@ -20,6 +20,7 @@
 package org.neo4j.graphdb;
 
 import java.util.Iterator;
+import java.util.function.Function;
```

Comment:
While i do think that 'java.util.function.Function' should be the only one function we should use, i do think that start using it without proper cleanup of old functions only adding yet another level of burden